compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -863,13 +865,24 @@ impl<'a> Parser<'a> {
 
     /// Parses constness: `const` or nothing.
     fn parse_constness(&mut self) -> Const {
-        if self.eat_keyword(kw::Const) {
+        if self.look_ahead(1, |t| t != &token::OpenDelim(DelimToken::Brace))
```

Comment:
leave a comment stating that this is to prevent const blocks from being parsed as a const item and then causing a parsing error